### PR TITLE
[PDI-18792] Performance of the "Write to log" step when the "Print he…

### PR DESCRIPTION
**Pull request by smmribeiro**
…aders" checkbox is activated 5x slower since 5.4 release

These changes are to improve the performance and responsiveness of the logging tab.

@pentaho-lmartins @ssamora @bcostahitachivantara @belxiors

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 16m 50s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1169**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7719/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: ui/src/main/java/org/pentaho/di/ui/spoon/trans/LogBrowser.java [hunk 1]
```diff
@@ -122,51 +123,54 @@ public void run() {
...
                       }
+
+                      text.setStyleRange( styleRange );
                     }
                   }
                 }
 
+                // The maximum size of the log buffer
+                //
+                int maxSize = Props.getInstance().getMaxNrLinesInLog() * 150;
```
Comment (ssamora) [inline ui/src/main/java/org/pentaho/di/ui/spoon/trans/LogBrowser.java:172]: Can the 150 be replaced by the newly created variable MAX_NR_LOG_LINES_CHUNK ?

Comment (smmribeiro) [reply ui/src/main/java/org/pentaho/di/ui/spoon/trans/LogBrowser.java]: It's a coincidence the '150' value as, here, it's referring to the estimated size of a log line.
I'll extract it to a new constant with a self-explaining name.

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 17m 51s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl ui
```
---
:ok_hand: All tests passed!
 
Tests run: **1169**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](http://wingman.pentaho.net/job/projects/job/pdi/job/PR-7719/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**